src/drivers/net/mod.rs

```diff
@@ -30,32 +51,64 @@ pub fn netwait_and_wakeup(handles: &[usize], millis: Option<u64>) {
 		}
 	}
 
-	NET_SEM.acquire(millis);
+	let mut reset_nic = false;
+
+	// every 20ms we check if the driver should be in the polling mode
+	while POLLING.swap(false, Ordering::SeqCst) == true {
+		reset_nic = true;
+
+		let core_scheduler = core_scheduler();
+		let wakeup_time = Some(crate::arch::processor::get_timer_ticks() + 20_000);
```

Comment:
Can we make that 20_000 configurable? Or at least move it to a constant for now? It may be interesting to try different values for some workloads just to be sure 20ms is an optimal value. Or is there any specific reason why this has to 20ms?

Reply:
I will do it